fenix/app/src/main/java/org/mozilla/fenix/onboarding/view/OnboardingScreen.kt

```diff
@@ -97,6 +99,12 @@ fun OnboardingScreen(
         }
     }
 
+    if (shouldShowDefaultBrowser &&
+        !pagesToDisplay.any { it.type == OnboardingPageUiData.Type.DEFAULT_BROWSER }
+    ) {
+        onMakeFirefoxDefaultClick()
+    }
```

Comment:
This logic should be in the Fragment. We should pass the list of only the screens we want to show to this composable. 

Another  issue I think could occur with this being in the composition will be called every time this recomposes.